### PR TITLE
JWT Auth impl

### PR DESCRIPTION
**Pull request by amrutamundkar**

### CODE REVIEW

**Comment by ajit-kulkarni** [comment]
@amrutamundkar Can you please add the API document as discussed ?

File: src/main/java/com/synerzip/VOs/EmployeeVO.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	
+/*	private LocalDateTime created_date;
+	
+	private String createdBy;
+
+	private LocalDateTime modified_date;
+	
+	private String modified_by;*/
+	
+	private Boolean is_active;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/VOs/EmployeeVO.java:24]: remove commented code

File: src/main/java/com/synerzip/VOs/EmployeeVO.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		return modified_by;
+	}
+
+
+
+	public void setModified_by(String modified_by) {
+		this.modified_by = modified_by;
+	}
+
+*/
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/VOs/EmployeeVO.java:127]: remove commented code

File: src/main/java/com/synerzip/auth/security/JWTAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                                                HttpServletResponse res) throws AuthenticationException {
+        try {
+            ApplicationUser creds = new ObjectMapper()
+                    .readValue(req.getInputStream(), ApplicationUser.class);
+
+            return authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(
+                            creds.getUsername(),
+                            creds.getPassword(),
+                            new ArrayList<>())
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/security/JWTAuthenticationFilter.java:49]: Remove empty arraylist

File: src/main/java/com/synerzip/auth/security/SecurityConstants.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.synerzip.auth.security;
+
+public class SecurityConstants {
+	public static final String SECRET = "SynerzipAM";
+    public static final long EXPIRATION_TIME = 864_000_000; // 10 days
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/security/SecurityConstants.java:5]: 864_000_000 is equal to one day

File: src/main/java/com/synerzip/auth/security/JWTAuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        if (token != null) {
+            // parse the token.
+            String user = Jwts.parser()
+                    .setSigningKey(SECRET.getBytes())
+                    .parseClaimsJws(token.replace(TOKEN_PREFIX, ""))
+                    .getBody()
+                    .getSubject();
+
+            if (user != null) {
+                return new UsernamePasswordAuthenticationToken(user, null, new ArrayList<>());
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/security/JWTAuthorizationFilter.java:57]: Get authorities from UserDetails, Do not pass empty arrayList

File: src/main/java/com/synerzip/auth/security/SecurityConstants.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.synerzip.auth.security;
+
+public class SecurityConstants {
+	public static final String SECRET = "SynerzipAM";
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/security/SecurityConstants.java:4]: Keep it synerzip only

File: src/main/java/com/synerzip/auth/serviceimpl/UserDetailsServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        this.applicationUserRepository = applicationUserRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        ApplicationUser applicationUser = applicationUserRepository.findByUsername(username);
+        if (applicationUser == null) {
+            throw new UsernameNotFoundException(username);
+        }
+        return new User(applicationUser.getUsername(), applicationUser.getPassword(), emptyList());
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/serviceimpl/UserDetailsServiceImpl.java:28]: do not pass empty list

File: src/main/java/com/synerzip/auth/usercontroller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+    public UserController(ApplicationUserRepository applicationUserRepository,
+                          BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.applicationUserRepository = applicationUserRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+
+    @PostMapping("/sign-up")
+    public void signUp(@RequestBody ApplicationUser user) {
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/usercontroller/UserController.java:27]: this should be done in service

File: src/main/java/com/synerzip/auth/usercontroller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        applicationUserRepository.save(user);
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/auth/usercontroller/UserController.java:28]: here , you should call service method

File: src/main/java/com/synerzip/controller/DashboardController.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+	
+	@GetMapping(value ="/budata")
+	public ResponseEntity<String> getData() {
+		
+		return new ResponseEntity<String>(userBU, HttpStatus.OK);
+	}
+	
+	//@CrossOrigin(origins = "http://localhost:3000")
+	@GetMapping(value="/data")
+	public ResponseEntity<Map> getEmployeeCount() {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/DashboardController.java:46]: return type should be ResponseEntity<Map<String,Long>>

File: src/main/java/com/synerzip/controller/DashboardController.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+	@Value("${synerzip.user.bu}")
+	private String userBU;
+	
+	@GetMapping(value ="/budata")
+	public ResponseEntity<String> getData() {
+		
+		return new ResponseEntity<String>(userBU, HttpStatus.OK);
+	}
+	
+	//@CrossOrigin(origins = "http://localhost:3000")
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/DashboardController.java:44]: remove commented code

File: src/main/java/com/synerzip/model/Role.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+@Entity
+@Table(name = "roles")
+public class Role {
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long role_id;
+	
+	@Column(nullable = false)
+	private String role_type;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Role.java:19]: make it roleName

File: src/main/java/com/synerzip/repository/EmployeeRepository.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import com.synerzip.model.Employee;
+
+public interface EmployeeRepository extends JpaRepository<Employee, Long> {
+/*
+	@Query(value = "select(SELECT COUNT(emp_id) as head_count from Employee),"
+			+"(SELECT COUNT(emp_id) as billable_count from BillingStatus where billable_status='billable' ),"
+			+"(SELECT COUNT(emp_id) as bench_count from BillingStatus where billable_status='bench')")	
+	List<DashboardDataVO> getDashboardData();
+	*/
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/repository/EmployeeRepository.java:17]: remove commented code

File: src/main/java/com/synerzip/service/BillingStatusService.java [hunk 14]
```diff
@@ -0,0 +1,9 @@
+package com.synerzip.service;
+
+public interface BillingStatusService {
+
+	public Long getStatusCount(String string);
+
+	//Long getBillableCount(String string);
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/service/BillingStatusService.java:7]: remove commented code

File: src/main/resources/application.properties [hunk 15]
```diff
@@ -0,0 +1,12 @@
+
+spring.jpa.database-platform=org.hibernate.dialect.MySQL5Dialect
+spring.jpa.generate-ddl=true
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.driverClassName=com.mysql.jdbc.Driver
+spring.datasource.url=jdbc:mysql://127.0.0.1:3306/employee_management?useSSL=false
+spring.datasource.username=root
+spring.datasource.password=Mundkar123
```
Comment (bhagyashreekadamsynerzip) [inline src/main/resources/application.properties:8]: change the password, keep it admin or root

File: src/main/resources/application.properties [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+spring.jpa.show-sql = true
+synerzip.user.bu= BU 9
+#flyway.locations=classpath:db/migration
+#flyway.setValidateOnMigrate= false;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/resources/application.properties:12]: remove this, as you are not using flyway

File: src/main/java/com/synerzip/controller/EmployeeController.java [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+@RequestMapping(value="empapi/v1")
+public class EmployeeController {
+
+
+	@Autowired
+	private EmployeeService employeeService;
+
+	@PostMapping(value="/employees")
+	public ResponseEntity<EmployeeVO> createStudent(@RequestBody EmployeeVO employeeVO) {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/EmployeeController.java:28]: change method name

File: src/main/java/com/synerzip/controller/EmployeeController.java [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+			return new ResponseEntity<EmployeeVO>(result, HttpStatus.OK);
+
+		} catch (Exception e) {
+			return new ResponseEntity<EmployeeVO>(new EmployeeVO(), HttpStatus.EXPECTATION_FAILED);
+		}
+	}
+
+	
+	@GetMapping(value="/employees/{id}")
+	public ResponseEntity<List<EmployeeVO>> getStudent(@PathVariable("id") long id) {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/EmployeeController.java:41]: change method name

File: src/main/java/com/synerzip/controller/EmployeeController.java [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+		EmployeeVO employeeVO = employeeService.getEmployee(id);
+		return new ResponseEntity (employeeVO, HttpStatus.OK);
+	}
+	
+	@GetMapping(value="/employees")
+	public ResponseEntity<List<EmployeeVO>> getStudents() {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/EmployeeController.java:47]: again change method name

File: src/main/java/com/synerzip/controller/EmployeeController.java [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+
+	}
+	
+	@DeleteMapping(value="/employees/{name}")
+	public ResponseEntity<List<EmployeeVO>> deleteStudentByName(@RequestParam(value="name") String name) {
+		List<EmployeeVO> empList = employeeService.deleteEmployeeByName(name);
+		return new ResponseEntity<List<EmployeeVO>>(empList, HttpStatus.OK);	
+
+	}
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/controller/EmployeeController.java:80]: change method names for all the methods

File: src/main/java/com/synerzip/model/BillingStatus.java [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+
+	public float getInvolvement() {
+		return involvement;
+	}
+
+	public void setInvolvement(float involvement) {
+		this.involvement = involvement;
+	}
+	
+	public static long getSerialversionuid() {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/BillingStatus.java:90]: no need of this getSerialversionuid()

File: src/main/java/com/synerzip/model/BillingStatus.java [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+	private float involvement;
+
+	
+	public long getId() {
+		return id;
+	}
+
+	public void setId(long id) {
+		this.id = id;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/BillingStatus.java:48]: id is auto incremented so no need to set it

File: src/main/java/com/synerzip/model/Employee.java [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+
+	public long getEmp_id() {
+		return emp_id;
+	}
+
+
+
+	public void setEmp_id(long emp_id) {
+		this.emp_id = emp_id;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:60]: emp_id is auto incremented, no need to set it

File: src/main/java/com/synerzip/model/Employee.java [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+
+@Entity
+@Table(name = "employees")
+public class Employee implements Serializable {
+	private static final long serialVersionUID = 1L;
+	
+	@Id	
+	@Column(name="emp_id")
+	@GeneratedValue(strategy=GenerationType.AUTO)
+	private long emp_id;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:25]: Use camelcase,It should be employeeId

File: src/main/java/com/synerzip/model/Employee.java [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+	
+	@Column(nullable = false)
+	private String empFirstName;
+	
+	@Column(nullable = true)
+	private String emp_middle_name;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:31]: either keep it as empMiddleName or employeeMiddleName

File: src/main/java/com/synerzip/model/Employee.java [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+	@Convert(converter = LocalDateTimeConverter.class)
+	private LocalDateTime modified_date;
+
+	
+	private String modified_by;
+	
+	private Boolean is_active;
+	
+	/*@Column(nullable = false)
+	private long manager_id;*/
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:50]: remove commented code

File: src/main/java/com/synerzip/model/Employee.java [hunk 27]
```diff
@@ -0,0 +1,190 @@
...
+	private String createdBy;
+
+	@Convert(converter = LocalDateTimeConverter.class)
+	private LocalDateTime modified_date;
+
+	
+	private String modified_by;
+	
+	private Boolean is_active;
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:48]: all variables should be in camelcase

File: src/main/java/com/synerzip/model/Employee.java [hunk 28]
```diff
@@ -0,0 +1,190 @@
...
+		return manager_id;
+	}
+
+
+
+	public void setManager_id(long manager_id) {
+		this.manager_id = manager_id;
+	}*/
+
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:169]: remove commented code

File: src/main/java/com/synerzip/model/Employee.java [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+
+	public static long getSerialversionuid() {
+		return serialVersionUID;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:173]: No need of this getSerialversionuid()

File: src/main/java/com/synerzip/model/Employee.java [hunk 30]
```diff
@@ -0,0 +1,190 @@
...
+	
+	
+	
+	
+	
+	
+	
+	
+
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Employee.java:190]: delete empty lines as well

File: src/main/java/com/synerzip/model/EmployeeRoleMapping.java [hunk 31]
```diff
@@ -0,0 +1,80 @@
...
+	private Employee employee;
+	
+	@ManyToOne(cascade = CascadeType.ALL)
+	@JoinColumn(name = "role", referencedColumnName = "role_id")	
+	private Role role;
+	
+	@ManyToOne(cascade = CascadeType.ALL)
+	@JoinColumn(name = "manager_id", referencedColumnName = "emp_id")
+	private Employee manager_id;
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/EmployeeRoleMapping.java:35]: variable names should be in camelcase

File: src/main/java/com/synerzip/model/ManagerialArea.java [hunk 32]
```diff
@@ -0,0 +1,41 @@
...
+@Table(name = "managerial_area")
+public class ManagerialArea {
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+	
+	@Column(nullable = false)
+	private String managerial_area_name;
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/ManagerialArea.java:20]: variable names shold be in camelcase

File: src/main/java/com/synerzip/model/ManagerialArea.java [hunk 33]
```diff
@@ -0,0 +1,41 @@
...
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/ManagerialArea.java:27]: no need to set id, as id is auto incremented

File: src/main/java/com/synerzip/model/Project.java [hunk 34]
```diff
@@ -0,0 +1,113 @@
...
+
+	@Convert(converter = LocalDateTimeConverter.class)
+	private LocalDateTime modified_date;
+
+	private String modified_by;
+	
+	@Column(nullable = false)
+	private Boolean is_active;
+
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Project.java:42]: variable names shold be in camelcase

File: src/main/java/com/synerzip/model/Project.java [hunk 35]
```diff
@@ -0,0 +1,113 @@
...
+	
+
+	public long getProject_id() {
+		return project_id;
+	}
+
+	public void setProject_id(Long project_id) {
+		this.project_id = project_id;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Project.java:50]: no need to set id as it is auto incremented

File: src/main/java/com/synerzip/model/Role.java [hunk 36]
```diff
@@ -0,0 +1,41 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "roles")
+public class Role {
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long role_id;
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Role.java:16]: variable names shold be in camelcase

File: src/main/java/com/synerzip/model/Role.java [hunk 37]
```diff
@@ -0,0 +1,41 @@
...
+
+	
+
+	public long getRole_id() {
+		return role_id;
+	}
+
+	public void setRole_id(Long role_id) {
+		this.role_id = role_id;
+	}
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Role.java:29]: no need to set Id

File: src/main/java/com/synerzip/model/Skill.java [hunk 38]
```diff
@@ -0,0 +1,39 @@
...
+@Table(name = "skills")
+public class Skill {
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long skill_id;
+	
+	@Column(nullable = false)
+	private String skill_name;	
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Skill.java:20]: variable names shold be in camelcase

File: src/main/java/com/synerzip/model/Skill.java [hunk 39]
```diff
@@ -0,0 +1,39 @@
...
+	private String skill_name;	
+	
+	public long getSkill_id() {
+		return skill_id;
+	}
+
+	public void setSkill_id(Long skill_id) {
+		this.skill_id = skill_id;
+	}
+
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/model/Skill.java:28]: no need to set id

File: src/main/java/com/synerzip/repository/EmployeeRepository.java [hunk 40]
```diff
@@ -0,0 +1,20 @@
...
+public interface EmployeeRepository extends JpaRepository<Employee, Long> {
+/*
+	@Query(value = "select(SELECT COUNT(emp_id) as head_count from Employee),"
+			+"(SELECT COUNT(emp_id) as billable_count from BillingStatus where billable_status='billable' ),"
+			+"(SELECT COUNT(emp_id) as bench_count from BillingStatus where billable_status='bench')")	
+	List<DashboardDataVO> getDashboardData();
+	*/
+	
+	List<Employee> findByempFirstNameContaining(String firstName);
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/repository/EmployeeRepository.java:18]: this is not used in the application, so remove it

File: src/main/java/com/synerzip/serviceImpl/BillingStatusServiceImpl.java [hunk 41]
```diff
@@ -0,0 +1,22 @@
...
+import com.synerzip.service.BillingStatusService;
+
+@Service
+public class BillingStatusServiceImpl implements BillingStatusService{
+	
+	@Autowired
+	BillingStatusRepository billingStatusRepository;
+
+	@Override
+	public Long getStatusCount(String string) {
```
Comment (bhagyashreekadamsynerzip) [inline src/main/java/com/synerzip/serviceImpl/BillingStatusServiceImpl.java:16]: give meaningful name to parameter

**Pull request closed without merging**